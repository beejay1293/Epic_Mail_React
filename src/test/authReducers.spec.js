import reducer, { initialState } from '../reducers/auth';
import {
  UNAUTHENTICATED,
  AUTHENTICATING,
  SIGNUP_ERROR,
  SIGNUP_SUCCESS,
  SIGNIN_ERROR,
  SIGNIN_SUCCESS,
  AUTHENTICATED,
} from '../constant/actionTypes.jsx';

describe('test auth reducers', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle AUTHENTICATIN', () => {
    expect(reducer(initialState, { type: AUTHENTICATING })).toMatchSnapshot();
  });


it('should handle AUTHETICATED', () => {
  expect(reducer(initialState, { type: AUTHENTICATED })).toMatchSnapshot();
})

it('should handle SIGNUP_SUCCESS', () => {
   expect(reducer(initialState, { type: SIGNUP_SUCCESS, payload: {data: {lastname: 'ji'}}})).toMatchSnapshot();
})

it('should handle SIGNIN_SUCCESS', () => {
   expect(reducer(initialState, { type: SIGNIN_SUCCESS, payload: {data: {lastname: 'ji'}}})).toMatchSnapshot();
})
it('should handle SINGUP_ERROR', () => {
  expect(reducer(initialState, {
    type: SIGNUP_ERROR
  })).toMatchSnapshot();
})

it('should handle SIGNIN_ERROR', () => {
  expect(reducer(initialState, {
    type: SIGNIN_ERROR
  })).toMatchSnapshot();
})

it('should handle UNAUTHENTICATED', () => {
  expect(reducer(initialState, {
    type: UNAUTHENTICATED
  })).toMatchSnapshot();
})


});