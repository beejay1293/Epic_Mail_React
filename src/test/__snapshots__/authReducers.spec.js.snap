// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test auth reducers should handle AUTHENTICATIN 1`] = `
Object {
  "errors": Object {},
  "isAuthenticated": false,
  "loading": true,
  "user": Object {},
  "working": false,
}
`;

exports[`test auth reducers should handle AUTHETICATED 1`] = `
Object {
  "errors": Object {},
  "isAuthenticated": true,
  "loading": false,
  "user": undefined,
  "working": false,
}
`;

exports[`test auth reducers should handle SIGNIN_ERROR 1`] = `
Object {
  "errors": undefined,
  "isAuthenticated": false,
  "loading": false,
  "user": Object {},
  "working": false,
}
`;

exports[`test auth reducers should handle SIGNIN_SUCCESS 1`] = `
Object {
  "errors": Object {},
  "isAuthenticated": true,
  "loading": false,
  "user": undefined,
  "working": false,
}
`;

exports[`test auth reducers should handle SIGNUP_SUCCESS 1`] = `
Object {
  "errors": Object {},
  "isAuthenticated": true,
  "loading": false,
  "user": undefined,
  "working": false,
}
`;

exports[`test auth reducers should handle SINGUP_ERROR 1`] = `
Object {
  "errors": undefined,
  "isAuthenticated": false,
  "loading": false,
  "user": Object {},
  "working": false,
}
`;

exports[`test auth reducers should handle UNAUTHENTICATED 1`] = `
Object {
  "errors": Object {},
  "isAuthenticated": false,
  "loading": false,
  "user": Object {},
  "working": false,
}
`;

exports[`test auth reducers should return the initial state 1`] = `
Object {
  "errors": Object {},
  "isAuthenticated": false,
  "loading": false,
  "user": Object {},
  "working": false,
}
`;
